### ISSUE TITLE
.net core 3

### ISSUE THREAD

**Issue by rafaelpirolla** [SiliconOrchid/RoslynRazorTemplating#3]
I was trying to use this on 3.0.101 on a Mac and got this:
```
Unhandled exception. System.Exception: (12,67): error CS0115: 'Template.ExecuteAsync()': no suitable method found to override
(14,13): error CS0103: The name 'WriteLiteral' does not exist in the current context
/Users/pirolla/Projects/RoslynRazorTemplating/src/EmailTemplateDemo/EmailTemplates/SampleEmailTemplate.razor(2,1): error CS0103: The name 'Write' does not exist in the current context
/Users/pirolla/Projects/RoslynRazorTemplating/src/EmailTemplateDemo/EmailTemplates/SampleEmailTemplate.razor(2,7): error CS0103: The name 'inherits' does not exist in the current context
(22,13): error CS0103: The name 'WriteLiteral' does not exist in the current context
/Users/pirolla/Projects/RoslynRazorTemplating/src/EmailTemplateDemo/EmailTemplates/SampleEmailTemplate.razor(6,12): error CS0103: The name 'Write' does not exist in the current context
/Users/pirolla/Projects/RoslynRazorTemplating/src/EmailTemplateDemo/EmailTemplates/SampleEmailTemplate.razor(6,18): error CS0103: The name 'Model' does not exist in the current context
(30,13): error CS0103: The name 'WriteLiteral' does not exist in the current context
/Users/pirolla/Projects/RoslynRazorTemplating/src/EmailTemplateDemo/EmailTemplates/SampleEmailTemplate.razor(11,13): error CS0103: The name 'Write' does not exist in the current context
/Users/pirolla/Projects/RoslynRazorTemplating/src/EmailTemplateDemo/EmailTemplates/SampleEmailTemplate.razor(11,19): error CS0103: The name 'Model' does not exist in the current context
(38,13): error CS0103: The name 'WriteLiteral' does not exist in the current context
/Users/pirolla/Projects/RoslynRazorTemplating/src/EmailTemplateDemo/EmailTemplates/SampleEmailTemplate.razor(15,35): error CS0103: The name 'Model' does not exist in the current context
(47,13): error CS0103: The name 'WriteLiteral' does not exist in the current context
/Users/pirolla/Projects/RoslynRazorTemplating/src/EmailTemplateDemo/EmailTemplates/SampleEmailTemplate.razor(18,32): error CS0103: The name 'Write' does not exist in the current context
(55,13): error CS0103: The name 'WriteLiteral' does not exist in the current context
(63,13): error CS0103: The name 'WriteLiteral' does not exist in the current context
   at RazorEngineTemplating.RazorEngineDynamicCompilerHelper`1.StreamAssemblyInMemory(CSharpCompilation cSharpCompilation) in /Users/pirolla/Projects/RoslynRazorTemplating/src/RazorEngineTemplating/RazorEngineDynamicCompilerHelper.cs:line 101
   at RazorEngineTemplating.RazorEngineDynamicCompilerHelper`1.CompileTemplate(String razorTemplateFileName, String dynamicAssemblyNamespace, String dynamicDllName) in /Users/pirolla/Projects/RoslynRazorTemplating/src/RazorEngineTemplating/RazorEngineDynamicCompilerHelper.cs:line 27
   at EmailTemplateDemo.Program.Main(String[] args) in /Users/pirolla/Projects/RoslynRazorTemplating/src/EmailTemplateDemo/Program.cs:line 28
```
Any clue on how to proceed would be great.

**Comment by SiliconOrchid** [comment]
Hey Rafael,  sorry things aren't working for you :-(

Let me see if I can help with some suggestions:-

My gut reaction, was that this is simply a missing method from `RazorEngineBaseTemplate`. However, that's not the problem as we can see that `WriteLiteral` method **is** included there ok.

I noticed that your output is identifying a `SampleEmailTemplate.razor` rather than `.cshtml`.  I've not tried renaming the file type, so I have no idea if this could change things - in theory it shouldn't make any difference.    However, if you've already been experimenting and changing the code away from the original sample, perhaps an unintended syntactic bug has been introduced which is preventing the template from being parsed?      I would recommend going back to basics and start by checking that the original sample code still works for you (i.e. cloning the repo afresh and try that)?

I've just cloned the project afresh, built it and run it  .... It's worked ok for me.  

Also, I currently have SDK 3.0.100 installed (windows), so we can probably discount this as being an SDK compatibility problem.   Running this code on a Mac shouldn't make any difference, but I haven't personally tested this outside of Windows so cannot confirm.

Let me know how you get on!
